theano/tensor/nnet/abstract_conv.py

```diff
@@ -2097,10 +2168,20 @@ def perform(self, node, inp, out_):
         axes_order = (1, 0) + tuple(range(2, self.convdim + 2))
         flip_filters = ((slice(None), slice(None)) +
                         (slice(None, None, -1),) * self.convdim)
+
+        def correct_for_groups(mat):
```

Comment:
The use of correct_for_groups is fine for the slow python version.
But as you wrote yourself, the transpose/reshape isn't optimal as it will cause a copy.
In the c code, we can just call the convolution many times and don't you have done that here? Anyway, it is slow python code, so I don't mind which version is used it. I just want to be sure the C code won't have that and using the same/more similar algo could be useful to help you make the change to the C code.